src/components/link/link.test.tsx

```diff
@@ -0,0 +1,570 @@
+
+  const linkElement = screen.getByTestId("link-anchor");
+
+  expect(linkElement).toHaveAttribute("target", target);
+});
+
+test("when component received a `rel` prop it should render an `<a>` element with rel attribute", () => {
+  const rel = "alternate";
+
+  render(<Link rel={rel} />);
+
+  const linkElement = screen.getByTestId("link-anchor");
+
+  expect(linkElement).toHaveAttribute("rel", rel);
+});
+
+test("when component receives a `href` prop it should render an `<a>` element with href attribute", () => {
+  const href = "https://www.google.com";
+  render(<Link href={href} />);
+
+  const linkElement = screen.getByRole("link");
+
+  expect(linkElement).toHaveAttribute("href", href);
+});
+
+test("should render an `Icon` correctly with the `basket` value", () => {
+  render(
+    <Link href="#test" icon="basket">
+      Test Content
+    </Link>
+  );
+
+  const iconElement = screen.getByTestId("icon");
+
+  expect(iconElement).toHaveAttribute("type", "basket");
+});
+
+test("should render an `Icon` on the left side of the component by default", () => {
+  render(<Link icon="basket" />);
+
+  const iconElement = screen.getByTestId("icon");
+
+  expect(iconElement).toHaveStyle({
+    marginRight: "var(--spacing050)",
+    position: "relative",
+  });
+});
+
+test("should render an `Icon` on the right", () => {
+  render(<Link icon="basket" iconAlign="right" />);
+
+  const iconElement = screen.getByTestId("icon");
+
+  expect(iconElement).toHaveStyle({
+    marginRight: "0",
+    marginLeft: "var(--spacing100)",
+    position: "relative",
+  });
+});
+
+test("should render an `Icon` on the right with no margin when no children", () => {
+  render(<Link icon="home" iconAlign="right" />);
+
+  const iconElement = screen.getByTestId("icon");
+
+  expect(iconElement).toHaveStyle({
+    marginRight: "0",
+    marginLeft: "0",
+    position: "relative",
+  });
+});
+
+test("when a link is rendered with an icon and no children, there should be no text decoration on the anchor element", () => {
+  render(<Link icon="home" href="www.sage.com" />);
+
+  const linkElement = screen.getByTestId("link-anchor");
+
+  expect(linkElement).toHaveStyle("text-decoration: none");
+});
+
+test("when a link is rendered with an icon and no children, link should have the inline display property", () => {
+  render(<Link icon="home" href="www.sage.com" />);
+
+  const iconElement = screen.getByTestId("icon");
+
+  expect(iconElement).toHaveStyle("display: inline");
+});
+
+test("when a link is rendered with an icon aligned right and has content, expected styles should be applied", () => {
+  render(
+    <Link icon="home" href="www.sage.com" iconAlign="right">
+      Has Content
+    </Link>
+  );
+
+  const iconElement = screen.getByTestId("icon");
+
+  expect(iconElement).toHaveStyle("margin-left: var(--spacing100)");
+});
+
+describe("when the `onKeyDown` event is triggered", () => {
+  it("should trigger an `onKeyDown` event", async () => {
```

Comment:
**suggestion**: might be worth re-wording this slightly so its a bit clearer?

```suggestion
  it("should call onKeyDown callback, when provided and link is triggered via the Enter key", async () => {
```

Reply:
I'll get this amended to what you have provided here. Thanks 😄